### DISCUSSION TITLE
How to prevent the TOC from being collapsed initially when I use chirpy-starter

### DISCUSSION THREAD

**Discussion by huanyushi** [cotes2020/jekyll-theme-chirpy#1706] [Q&A] [answered]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Generated from `chirpy-starter`

### Description

All TOC (tables of contents) are initially collapsed, and sub-items are uncollapsed only when the corresponding section is read. How can I cancel the folding of the TOC initially? Any suggestions would be greatly appreciated.

### Operations you have already tried

This question was solved when the site is generated from `jekyll-theme-chirpy`, we only need to modify the  file  `_javascript\modules\components\toc.js`.  The configurations of tocbot can be seen [here](https://github.com/tscanlin/tocbot?tab=readme-ov-file#options).

But this solution can not take effect when the site is generated from `chirpy-starter`, because if we creat a file `_javascript\modules\components\toc.js`, it can not override the theme's javascript file. 

### Anything else?

Relevent links: #1616 , #1364 .

**Comment by huanyushi** [comment, marked as answer]
Add the following code to "assets/css/jekyll-theme-chirpy.scss" can solve this problem:

```scss
.is-collapsed {
  max-height: none !important;
}
```

**Reply by HyoYoonNam** [reply]
This solution of modifying the CSS code works great!

But I have an additional question, if anyone can help.

My initial approach was to prevent the TOC from collapsing by modifying the tocbot options parameter in the _javascript/modules/components/toc/toc-desktop.js file.

Specifically, I added collapseDepth: 6, to the options object, which is the method described in the official [tocbot api](https://github.com/tscanlin/tocbot#usage).
<img width="690" height="329" alt="스크린샷 2025-08-17 22 05 24" src="https://github.com/user-attachments/assets/b636f24c-ec02-4c64-8780-c46346fb51ee" />


However, this approach didn't solve the issue for me. Does anyone know why this might be?
